src/main/webapp/linkWizard/SelectSampleConsent.js

```diff
@@ -109,6 +114,71 @@ export const SelectSampleConsent = hh(class SelectSampleConsent extends Componen
     this.setState({ documentOptions: documentOptions });
   };
 
+  getConsentGroups = () => {
+    this.setState({ consentGroupIsLoading: true });
+    ConsentGroup.getConsentGroupNames(this.props.getConsentGroups).then(
+      resp => {
+        const existingConsentGroups = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.label,
+            label: item.label
+          }
+        });
+
+        this.setState({
+          existingConsentGroups: existingConsentGroups,
+          consentGroupIsLoading: false,
+          consentGroup: existingConsentGroups[0]
+        }, () => this.props.updateForm(this.state.consentGroup, "consentGroup"));
+        this.getAllSampleCollections(existingConsentGroups[0].key);
+      }
+    );
+  };
+
+  getAllSampleCollections = (consentKey) => {
+    this.setState({ sampleCollectionIsLoading: true });
+
+    SampleCollections.getCollectionsCGLinked(this.props.getConsentGroupSampleCollections, consentKey).then(
+      resp => {
+        const sampleCollectionList = [];
+        sampleCollectionList.push({label: "Sample Collections Linked to " + consentKey, options: []});
+
+        sampleCollectionList[0].options = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.collectionId,
+            label: item.collectionId + ": " + item.name + " ( " + item.category + " )",
+          };
+        });
+
+        this.setState({
+          sampleCollectionList: sampleCollectionList,
+          sampleCollectionIsLoading: false
+        })
+      }
+    );
+
+    SampleCollections.getSampleCollections(this.props.unConsentedSampleCollections, consentKey).then(
+      resp => {
+        const sampleCollectionList = this.state.sampleCollectionList.splice(0);
+        sampleCollectionList.push({label: "Link New Sample Collections to Sample Data/Cohort: " + consentKey, options: []});
+        sampleCollectionList[1].options = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.collectionId,
+            label: item.collectionId + ": " + item.name + " ( " + item.category + " )",
+          };
+        });
```

Comment:
This block of code is almost identical to the one above it. The select option builder code should be pulled out into a utility function that expects a label and a collection, builds a new list of options, and returns that for the `setState` call.

Reply:
Yes, I agree.